From 9ec77cc982d3a3fb7bec436797258f6634063955 Mon Sep 17 00:00:00 2001
From: Brian Delwiche <delwiche@google.com>
Date: Tue, 2 Jul 2024 17:27:29 +0000
Subject: [PATCH] Remove flags for b#314331379

bluffs_mitigation has passed soak and flags for it should be removed.

Bug: 314331379
Bug: 319541706
Flag: bluffs_mitigation
Test: m libbluetooth
Tag: #security
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:9fc2e1e6d449e2665fc5d5d9af4ca87346863960)
(cherry picked from https://android-review.googlesource.com/q/commit:34a5ceced396f40eae30747d295e07d31f18e59d)
Merged-In: I2f2743e1bec281823cd0ca4cb67e281f3aa1df48
Change-Id: I2f2743e1bec281823cd0ca4cb67e281f3aa1df48
---
 system/stack/btm/btm_sec.cc | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/system/stack/btm/btm_sec.cc b/system/stack/btm/btm_sec.cc
index 08e2a3371e..4a83501c63 100644
--- a/system/stack/btm/btm_sec.cc
+++ b/system/stack/btm/btm_sec.cc
@@ -3475,7 +3475,13 @@ void btm_sec_encryption_change_evt(uint16_t handle, tHCI_STATUS status, uint8_t
               handle, base::Bind(&read_encryption_key_size_complete_after_encryption_change));
       return;
     }
-  }
+    btm_acl_encrypt_change(handle, static_cast<tHCI_STATUS>(status), encr_enable);
+    btm_sec_encrypt_change(handle, static_cast<tHCI_STATUS>(status), encr_enable, 0);
+  } else {
+    btsnd_hcic_read_encryption_key_size(
+      handle,
+      base::Bind(&read_encryption_key_size_complete_after_encryption_change));
+  }  
 
   if (status == HCI_ERR_CONNECTION_TOUT) {
     smp_cancel_start_encryption_attempt();
@@ -3489,8 +3495,6 @@ void btm_sec_encryption_change_evt(uint16_t handle, tHCI_STATUS status, uint8_t
                          "stack::btu::btu_hcif::encryption_change_evt Encryption Failure");
     }
   }
-  btm_acl_encrypt_change(handle, static_cast<tHCI_STATUS>(status), encr_enable);
-  btm_sec_encrypt_change(handle, static_cast<tHCI_STATUS>(status), encr_enable, 0);
 }
 /*******************************************************************************
  *
-- 
2.34.1

